pkg/receiver/rawk8seventsreceiver/receiver.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rawk8seventsreceiver
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	backoff "github.com/cenkalti/backoff/v4"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Only two types of events are created as of now.
+// For more info: https://docs.openshift.com/container-platform/4.9/rest_api/metadata_apis/event-core-v1.html
+var severityMap = map[string]pdata.SeverityNumber{
+	"normal":  pdata.SeverityNumberINFO,
+	"warning": pdata.SeverityNumberWARN,
+}
+
+type rawK8sEventsReceiver struct {
+	cfg              *Config
+	client           k8s.Interface
+	eventControllers []cache.Controller
+	eventCh          chan *corev1.Event
+	ctx              context.Context
+	cancel           context.CancelFunc
+	startTime        time.Time
+
+	consumer consumer.Logs
+	logger   *zap.Logger
+}
+
+// Interface for creating ListerWatcher objects. Used for injecting mocks into k8s informers.
```

Comment:
It's not an interface but a type 🙃

Reply:
It's colloquially an interface, because afaik you can't define an actual callable interface in Go. I can change the wording if you think it's confusing.

Reply:
> It's colloquially an interface

Don't know what that means 😆 

> afaik you can't define an actual callable interface in Go. 

Because that's not a thing. You call functions, you fulfill interfaces.